File: amtc_common/test/test_NodeConfigure.cpp
/*
 * test_Communications.cpp
 *
 *  Created on: May 31, 2014
 *      Author: isao
 */
#include <amtc_common/Communications.h>
#include <std_msgs/String.h>
#include <std_srvs/SetBool.h>
amtc::Communications* comms_ptr;

void timer_cb(const ros::TimerEvent& e)
{
  ROS_FATAL("TEST");
}


int main(int argc, char** argv)
{
  ros::init(argc, argv, "test_NodeConfigure");
  ros::NodeHandle n("~");

  comms_ptr = new amtc::Communications(n);

  comms_ptr->getNodeConfigure().initialize();

  comms_ptr->add_timer("timer",ros::Duration(comms_ptr->getNodeConfigure().get_local_param<double>("timer_period")),timer_cb);

  {
    std::vector<std::string> list;
    comms_ptr->getNodeConfigure().get_named_action_topic_list(list);
    ROS_FATAL("ACTION_TOPIC");
    BOOST_FOREACH(std::string topic, list)
    {
      ROS_FATAL("\t%s : %s", topic.c_str(), comms_ptr->getNodeConfigure().get_named_action_topic(topic).c_str());
    }
  }
  {
    std::vector<std::string> list;
    comms_ptr->getNodeConfigure().get_named_topic_list(list);
    ROS_FATAL("TOPIC");
    BOOST_FOREACH(std::string topic, list)
    {
      ROS_FATAL("\t%s : %s", topic.c_str(), comms_ptr->getNodeConfigure().get_named_topic(topic).c_str());
    }
  }
  {
    std::vector<std::string> list;
    comms_ptr->getNodeConfigure().get_named_service_list(list);
    ROS_FATAL("SERVICE");
    BOOST_FOREACH(std::string element, list)
    {
      ROS_FATAL("\t%s : %s", element.c_str(), comms_ptr->getNodeConfigure().get_named_service(element).c_str());
    }
  }
  {
    std::vector<std::string> list;
    comms_ptr->getNodeConfigure().get_named_node_list(list);
    ROS_FATAL("NODE");
    BOOST_FOREACH(std::string element, list)
    {
      ROS_FATAL("\t%s : %s", element.c_str(), comms_ptr->getNodeConfigure().get_named_node(element).c_str());
    }
  }
  {
    std::vector<std::string> list;
    comms_ptr->getNodeConfigure().get_named_namespace_list(list);
    ROS_FATAL("NAMESPACE");
    BOOST_FOREACH(std::string element, list)
    {
      ROS_FATAL("\t%s : %s", element.c_str(), comms_ptr->getNodeConfigure().get_named_namespace(element).c_str());
    }
  }
  {
    ROS_FATAL("GLOBAL_PARAM");
    std::string element;
    element = "name_topic";
    ROS_FATAL("\t%s : %s", element.c_str(), comms_ptr->getNodeConfigure().get_global_param<std::string>(element).c_str());
    element = "timeout";
    ROS_FATAL("\t%s : %f", element.c_str(), comms_ptr->getNodeConfigure().get_global_param<double>(element));
    element = "sequence";
    std::vector<double> double_list(comms_ptr->getNodeConfigure().get_global_param<std::vector<double>>(element));
    ROS_FATAL("\t%s : ", element.c_str());
    BOOST_FOREACH(double value, double_list)
    {
      ROS_FATAL("\t\t%f", value);
    }
  }
  {
    ROS_FATAL("LOCAL_PARAM");
    std::string element;
    element = "delay_start";
    ROS_FATAL("\t%s : %f", element.c_str(), comms_ptr->getNodeConfigure().get_local_param<double>(element));
    element = "spin_time";
    ROS_FATAL("\t%s : %f", element.c_str(), comms_ptr->getNodeConfigure().get_local_param<double>(element));
    element = "timer_period";
    ROS_FATAL("\t%s : %f", element.c_str(), comms_ptr->getNodeConfigure().get_local_param<double>(element));
  }



  while(ros::ok())
  {
	  comms_ptr->spin();
  }
}

File: amtc_common/test/test_NodeConfigure.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-



import rospy
import std_msgs.msg
import std_srvs.srv
import math
import amtc_common.Communications as amtc



class TestCommunications():
  def __init__(self):
    rospy.init_node('test_NodeConfigure')
    self.comms = amtc.Communications()
    self.comms.node_configure.initialize()


    aux_list  = self.comms.node_configure.get_named_action_topic_list()
    rospy.logfatal('ACTION TOPIC')
    rospy.logfatal('\t'+str(aux_list))
    rospy.logfatal('\n')

    aux_list  = self.comms.node_configure.get_named_topic_list()
    rospy.logfatal('TOPIC')
    rospy.logfatal('\t'+str(aux_list))
    rospy.logfatal('\n')

    aux_list  = self.comms.node_configure.get_named_service_list()
    rospy.logfatal('SERVICE')
    rospy.logfatal('\t'+str(aux_list))
    rospy.logfatal('\n')

    aux_list  = self.comms.node_configure.get_named_node_list()
    rospy.logfatal('NODE')
    rospy.logfatal('\t'+str(aux_list))
    rospy.logfatal('\n')

    aux_list  = self.comms.node_configure.get_named_namespace_list()
    rospy.logfatal('NAMESPACE')
    rospy.logfatal('\t'+str(aux_list))
    rospy.logfatal('\n')

    rospy.logfatal('GLOBAL_PARAM')
    element_list= ['name_topic', 'timeout', 'sequence']
    for element in element_list:
      rospy.logfatal('\t'+str(element)+' : '+ str(self.comms.node_configure.get_global_param(element)))
      pass
    rospy.logfatal('\n')

    rospy.logfatal('LOCAL_PARAM')
    element_list= ['delay_start', 'spin_time', 'timer_period']
    for element in element_list:
      rospy.logfatal('\t'+str(element)+' : '+ str(self.comms.node_configure.get_local_param(element)))
      pass
    rospy.logfatal('\n')



    pass

  def input_cb(self,msg):
    print ("income: " +msg.data)
    pass

  def timer_cb(self,event):
    self.comms.get_publisher('string_pub').get_msg().data = 'HOLA'
    self.comms.publish('string_pub')
    self.comms.get_service_client('service_client').wait_for_service()
    req = std_srvs.srv.SetBoolRequest()
    req.data = True
    resp = self.comms.call_service('service_client', req)
    self.comms.get_publisher('string_pub').get_msg().data = resp.message
    self.comms.publish('string_pub')

    pass

  def service_server_cb(self,request):
    print ("service_server: "+str(request.data))
    resp = std_srvs.srv.SetBoolResponse()
    resp.message = 'OK service'
    resp.success = True
    return resp
    pass


  def run(self):
    rospy.spin()
    pass

pass


if __name__ == '__main__':
    node = TestCommunications()
    node.run()


